app/javascript/packs/validateFileInputs.js

```diff
@@ -0,0 +1,192 @@
+/**
```

Comment:
The JS Docs comments explain a lot.

An additional clarification is the default error handling is to add red text with the error message to the form field and clear the form field.

I added functionality to accept a custom callback to execute error handling by error type (size or format), but I didn't end up using it. Once I saw the default red text actually didn't display as poorly as I thought it would, I just stuck to the default. Some of this code I designed with the thinking in mind we could cut and paste chunks of it out into Active Storage event triggers if/when we go that direction.